test/rules/promise-async/test.ts.lint

```diff
@@ -0,0 +1,38 @@
+const nonAsyncPromiseFunction = function(p: Promise<void>) { return p; };
+                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    [0]
+
```

Comment:
also need cases 

```
x.callback(() => return new Promise<void>());
```

and 

```

function x() {
    return new Promise<int>();
}
```

which hit the new `visit*` cases I suggested above